src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1028 @@
+                out_past_key_values += (present,)
+            if self.config.output_attentions or output_attentions:
+                outpos += 1
+                attention_probs = block_output[outpos]
+                out_attentions += (attention_probs,)
+            if output_router_tuple:
+                outpos += 1
+                router_tuple = block_output[outpos]
+                out_router_logits.append(router_tuple[0])
+
+        hidden = self.logits(hidden)
+        hidden = self.logact(hidden)
+
+        logits = torch.einsum("bsc,vc->bsv", hidden, self.embed_tokens.weight)
+        if logits.shape[-1] == self.token_bias.shape[-1]:
+            logits = logits + self.token_bias
+
+        loss = None
+        if labels is not None:
+            loss_fct = nn.CrossEntropyLoss(ignore_index=-100)
+            loss = loss_fct(logits.view(-1, logits.size(-1)), labels.view(-1))
+
+        ret = ModelOutput()
+        logits[:, :, SEP_TOKEN] = -1e10
+        logits[:, :, NOT_TOKEN] = -1e10
+        ret["logits"] = logits
+        if self.config.output_hidden_states or output_hidden_states:
+            ret["hidden_states"] = out_hidden_states
+        if self.config.use_cache or use_cache:
+            ret["past_key_values"] = out_past_key_values
+        if self.config.output_attentions or output_attentions:
+            ret["attentions"] = out_attentions
+        if self.config.output_router_logits or output_router_logits:
+            ret["router_logits"] = out_router_logits
+        ret["loss"] = loss
+
+        if return_dict:
+            return ret
+
+        outp = collections.namedtuple("GPTSANOutputs", " ".join(ret.keys()))
+        return outp(**ret)
+
+    def prepare_inputs_for_generation(
+        self,
+        input_ids: torch.LongTensor,
+        connected_inputs: Optional[torch.LongTensor] = None,
+        spout: Optional[torch.FloatTensor] = None,
+        past_key_values: Optional[Tuple[Tuple[torch.FloatTensor]]] = None,
+        random_seed: Optional[int] = None,
+        **kwargs
+    ):
+        if past_key_values is not None:
+            return {
+                "input_ids": input_ids,
+                "num_precontext": connected_inputs,
+                "spout": None,
+                "past_key_values": past_key_values,
+            }
+        NUM_TOKENS = self.config.vocab_size
+        SOT_TOKEN = NUM_TOKENS - 7
+        SEG_TOKEN = NUM_TOKENS - 2
+        if connected_inputs is None:
+            connected_inputs = torch.zeros(input_ids.shape[0]).int().to(input_ids.device)
+        pre_input = torch.stack(
+            [
+                torch.tensor(
+                    [SOT_TOKEN]
+                    + input_ids[i, : connected_inputs[i]].cpu().numpy().tolist()
+                    + [SEG_TOKEN]
+                    + input_ids[i, connected_inputs[i] :].cpu().numpy().tolist()
+                )
+                for i in range(input_ids.shape[0])
+            ]
+        )
```

Comment:
same here. If the generation needs to start with some tokens, the tokenizer should be adding them automatically. You can check what is done for `Whisper` to have an example or `MBart`